src/txn/txn_rollback_to_stable.c

```diff
@@ -98,31 +98,38 @@ err:	if (ret == 0) {
  */
 static void
 __txn_abort_newer_update(WT_SESSION_IMPL *session,
-    WT_UPDATE *upd, wt_timestamp_t *rollback_timestamp)
+    WT_UPDATE *first_upd, wt_timestamp_t *rollback_timestamp)
 {
-	WT_UPDATE *next_upd;
-	bool aborted_one;
+	WT_UPDATE *upd;
+	bool skip_zero_timestamps;
+
+	skip_zero_timestamps = !FLD_ISSET(S2BT(session)->assert_flags,
+	    WT_ASSERT_COMMIT_TS_ALWAYS | WT_ASSERT_COMMIT_TS_KEYS);
 
-	aborted_one = false;
-	for (next_upd = upd; next_upd != NULL; next_upd = next_upd->next) {
+	for (upd = first_upd; upd != NULL; upd = upd->next) {
 		/*
-		 * Updates with no timestamp will have a timestamp of zero
-		 * which will fail the following check and cause them to never
-		 * be aborted.
+		 * Updates with no timestamp will have a timestamp of zero and
+		 * will never be rolled back.  If the table is configured for
+		 * strict timestamp checking, assert that we never have such
+		 * updates in the list ahead of updates we do roll back.
```

Comment:
Minor nit: I find `ahead` really confusing because I can interpret it two ways and I don't know which you mean. Do you mean ahead of `upd` that is walking the list and we haven't gotten to it yet (i.e. older) or ahead as in earlier in the list (i.e. younger)? Can you use `earlier/later in the list` or `younger/older`?